### PR TITLE
Add Node - pnpm example

### PR DESCRIPTION
**Pull request by ebeloded**

### CODE REVIEW

File: examples.md [hunk 1]
```diff
@@ -240,6 +240,17 @@ The yarn cache directory will depend on your operating system and version of `ya
 ```
 
 
+## Node - Pnpm
+
+```yaml
+  - name: Cache .pnpm-store
+    uses: actions/cache@v2
+    with:
+      path: ~/.pnpm-store
```
Comment (aminya) [inline examples.md:249]: The path is different on Windows.

```suggestion
      path: |
      	 ~/.pnpm-store
         D:\.pnpm-store
```

**Review by aminya — CHANGES_REQUESTED** [review]

File: examples.md (hunk 1 above)
Comment (ebeloded) [reply examples.md]: Thank you!

File: examples.md [hunk 2]
```diff
@@ -277,6 +277,20 @@ The yarn cache directory will depend on your operating system and version of `ya
...
+## Node - Pnpm
+
+```yaml
+  - name: Cache .pnpm-store
+    uses: actions/cache@v2
+    with:
+      path: |
+      	 ~/.pnpm-store
+         D:\.pnpm-store
+      key: ${{ runner.os }}-${{ hashFiles('**/pnpm-lock.yaml') }}```
```
Comment (aminya) [inline examples.md:290]: `pnpm-store` is mostly independent of `pnpm-lock.yaml`, and restoring the store even when the lock file has changed can save a lot of time

```suggestion
      key: ${{ runner.os }}```
```

**Review by Limlay — APPROVED** [review]

**Review by Limlay — APPROVED** [review]

**Review by Limlay — APPROVED** [review]

**Review by aminya — APPROVED** [review]

**Comment by ebeloded** [comment]
@aminya @Limlay 
Since last review I have added `**/node_modules` to the list of cached resources. This allows to cache artifacts that are generated during post install step. With this change the `pnpm install` script is instantaneous.

File: examples.md [hunk 3]
```diff
@@ -277,6 +277,22 @@ The yarn cache directory will depend on your operating system and version of `ya
...
+
+## Node - Pnpm
+
+```yaml
+  - name: Cache .pnpm-store
+    uses: actions/cache@v2
+    with:
+      path: |
+      	 ~/.pnpm-store
+         D:\.pnpm-store
```
[the remark this answers is not in the document]
Comment (aminya) [reply examples.md:290]: The official pnpm setup installs pnpm in the D drive. Also, it doesn't hurt if the path doesn't exist. The users can update this later.

**Review by beary — APPROVED** [review]

**Review by beary — APPROVED** [review]

**Review by beary — APPROVED** [review]

File: examples.md [hunk 4]
```diff
@@ -277,6 +277,22 @@ The yarn cache directory will depend on your operating system and version of `ya
...
+      key: ${{ runner.os }}-${{ hashFiles('**/pnpm-lock.yaml') }}
+      restore-keys: |
+        ${{ runner.os }}
```
[the remark this answers is not in the document]
Comment (ebeloded) [reply examples.md]: Added `-pnpm-` in cache key

File: examples.md [hunk 5]
```diff
@@ -17,6 +17,7 @@
   - [Windows](#windows-2)
   - [Using multiple systems and `npm config`](#using-multiple-systems-and-npm-config)
 - [Node - Lerna](#node---lerna)
+- [Node - Yarn](#node---pnpm)
```
Comment (orta) [inline examples.md:20]: Typo:

```suggestion
- [Node - pnpm](#node---pnpm)
```

**Comment by hckhanh** [comment]
Hi @orta,
Is there any updates for this PR?

**Comment by orta** [comment]
I'm afraid I don't have write access, I believe it's on @tiwarishub to make the call - looks reasonable to me though

**Comment by karlhorky** [comment]
@joshmgross what do you think about adding the changes from this PR?

I think it was closed because of lack of response from reviewers...

File: examples.md [hunk 6]
```diff
@@ -353,6 +354,23 @@ The yarn cache directory will depend on your operating system and version of `ya
...
+
+## Node - Pnpm
+
+```yaml
+  - name: Cache .pnpm-store
+    uses: actions/cache@v2
+    with:
+      path: |
+      	 ~/.pnpm-store
+         D:\.pnpm-store
```
Comment (karlhorky) [inline examples.md:366]: these pnpm store paths could be resolved from running `pnpm store path --silent`, as is done in `actions/setup-node`:

- https://github.com/actions/setup-node/blob/26961cf329f22f6837d5f54c3efd76b480300ace/src/cache-utils.ts#L35

**Comment by karlhorky** [comment]
Probably a better version of the PR would be this, taken from [the last version in the pnpm readme](https://github.com/pnpm/action-setup/pull/131)

````markdown
## Node - pnpm

```yaml
      - name: Get pnpm store directory
        shell: bash
        run: |
          echo "STORE_PATH=$(pnpm store path --silent)" >> $GITHUB_ENV
      - uses: actions/cache@v4
        name: Setup pnpm cache
        with:
          path: ${{ env.STORE_PATH }}
          key: ${{ runner.os }}-pnpm-store-${{ hashFiles('**/pnpm-lock.yaml') }}
          restore-keys: |
            ${{ runner.os }}-pnpm-store-
```
````

**Comment by karlhorky** [comment]
Opened a followup PR here:

- https://github.com/actions/cache/pull/1451

**Pull request closed without merging**